test/unit/logger-client.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+// Testing env vars is painful here. Override the default and check config-driven
+// option later
+process.env.LOGGER_SEND_USER_AGENT = ''
+
+const {test} = require('tap')
+const pkg = require('../../package.json')
+const loggerClient = require('../../lib/logger-client.js')
+const config = require('../../lib/config.js')
+
+const CONFIG_DEFAULT = {
+  ...config
+, key: 'abc123'
+, UserAgent: 'fake-transport@1.2.3'
+, hostname: 'myhostname'
+}
+
+test('Check logger defaults', async (t) => {
+  const logger = loggerClient.createLoggerClient(CONFIG_DEFAULT)
+  const requestDefaults = logger[Symbol.for('requestDefaults')]
+
+  t.match(requestDefaults.headers, {
+    'Content-Type': 'application/json; charset=UTF-8'
+  , 'Content-Encoding': 'gzip'
+  }, 'Default headers are correct')
+
+  const expectedAgentHeader = `@logdna/logger/${
+    pkg.dependencies['@logdna/logger'].replace('^', '')
```

Comment:
technically the semver in package.json might differ from the version in the package-lock.json. probably a non-issue in practice, but in theory you _could_ instead swap the require to be `package-lock.json` and always get the true resolved version.

Reply:
Ah, good point.  I think just making it a regex would suffice.  It feels like making it match exact is introducing unnecessary complexity.

Reply:
done